src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowLoopSpec.groovy

```diff
@@ -0,0 +1,618 @@
+//        assert northbound.getSwitchRules(sw.dpId).flowEntries.findAll {
+//            new Cookie(it.cookie).getType() == CookieType.FLOW_LOOP
+//        }.size() == 2
+//    }
+
+        and: "The switch is valid"
+        //there is a bug, flow_loop is in excess
+//      northbound.validateSwitch(sw.dpId).verifyRuleSectionsAreEmpty(["missing", "excess", "misconfigured"])
+
+        when: "Delete the flow with created flowLoop"
+        flowHelperV2.deleteFlow(flow.flowId)
+        def flowIsDeleted = true
+
+        then: "FlowLoop rules are deleted on the switch"
+        assert northbound.getSwitchRules(sw.dpId).flowEntries.findAll {
+            new Cookie(it.cookie).getType() == CookieType.FLOW_LOOP
+        }.empty
+
+        and: "The switch is valid"
+        northbound.validateSwitch(sw.dpId).verifyRuleSectionsAreEmpty(["missing", "excess", "misconfigured"])
+        def testIsCompleted = true
+
+        cleanup: "Delete the flow"
+        !flowIsDeleted && flowHelperV2.deleteFlow(flow.flowId)
+        !testIsCompleted && northbound.synchronizeSwitch(sw.dpId, true)
+    }
+
+    @Tidy
+    def "System is able to reroute a flow when flowLoop is created on it"() {
+        given: "A multi switch flow with one alternative path at least"
+        def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find {
+            it.paths.unique(false) {
+                a, b -> a.intersect(b) == [] ? 1 : 0
+            }.size() >= 2
+        }
+        def flow = flowHelperV2.randomFlow(switchPair)
+        flowHelperV2.addFlow(flow)
+        def flowPath = PathHelper.convert(northbound.getFlowPath(flow.flowId))
+
+        and: "FlowLoop is created on the dst switch"
+        northboundV2.createFlowLoop(flow.flowId, new FlowLoopPayload(switchPair.dst.dpId))
+
+        when: "Fail a flow ISL (bring switch port down)"
+        def allFlowPaths = switchPair.paths
+        Set<Isl> altFlowIsls = []
+        def flowIsls = pathHelper.getInvolvedIsls(flowPath)
+        allFlowPaths.findAll { it != flowPath }.each { altFlowIsls.addAll(pathHelper.getInvolvedIsls(it)) }
+        def islToFail = flowIsls.find { !(it in altFlowIsls) && !(it.reversed in altFlowIsls) }
+        antiflap.portDown(islToFail.srcSwitch.dpId, islToFail.srcPort)
+
+        then: "The flow was rerouted"
+        Wrappers.wait(rerouteDelay + WAIT_OFFSET) {
+            assert northboundV2.getFlowStatus(flow.flowId).status == FlowState.UP
+            assert PathHelper.convert(northbound.getFlowPath(flow.flowId)) != flowPath
+        }
+
+        and: "Flow is UP and valid"
+        Wrappers.wait(WAIT_OFFSET) { assert northbound.getFlowStatus(flow.flowId).status == FlowState.UP }
+        northbound.validateFlow(flow.flowId).each { direction -> assert direction.asExpected }
+
+        and: "FlowLoop is still created on the src switch"
+        with(northbound.getFlow(flow.flowId)) {
+            it.looped
+            it.loopSwitchId == switchPair.dst.dpId
+        }
```

Comment:
I think it makes sense to check that loop rules are indeed still present on the switch